editor/src/messages/portfolio/document/graph_operation/graph_operation_message_handler.rs

```diff
@@ -701,11 +701,21 @@ impl MessageHandler<GraphOperationMessage, GraphOperationMessageData<'_>> for Gr
 			}
 			GraphOperationMessage::SetNameImpl { layer, name } => {
 				if let Some(node) = document_network.nodes.get_mut(&layer.to_node()) {
-					node.alias = name;
-					node_graph.update_click_target(layer.to_node(), document_network, Vec::new());
-					responses.add(DocumentMessage::RenderRulers);
-					responses.add(DocumentMessage::RenderScrollbars);
-					responses.add(NodeGraphMessage::SendGraph);
+					let node_id = layer.to_node();
+
+					if node.name == "Artboard" {
+						let input_index = 2;
+						let value = TaggedValue::String(name);
+
+						responses.add(NodeGraphMessage::SetInputValue { node_id, input_index, value });
```

Comment:
If the NodeInput for the String is exposed and connected to another node with NodeInput::Node, this will disconnect it when the name is changed in the layer panel. It doesn't really make sense to expose the name as an Input, since the only way it should be modified is through modifying the `DocumentNode` alias property. 

I think it is best to remove the NodeInput entirely, and automatically add a NodeInput::Value with the `TaggedValue::String` based on the DocumentNode alias. This can be done in the `flatten_with_fns` step of the compilation process, where the input could be added when the `NodeNetwork` is compiled into an executable `ProtoNetwork`. This will allow the input to be fully hidden, similar to how footprints are automatically added. (This is quite advanced, but also a great way to learn about the compilation process. I can help more if I have time)

Reply:
Thanks for the feedback. I will try to follow what you said here. Is there a place where I can learn more about the way node graph works? Or maybe an overall diagram that explains how things come together. 

And, not related but, one more question. Is there a way to speed up build process? Every time I save a file, wasm compiles again and it takes too much time.

Reply:
I outlined a bit of the process here, and the linked discord threads have more information
https://github.com/GraphiteEditor/Graphite/issues/1762. Id recommend tracing over the entire compilation process, which starts [here](https://github.com/GraphiteEditor/Graphite/blob/open-sauce/editor/src/node_graph_executor.rs#L216) to get a basic understanding.

 Building will always be slow due to wasm-pack and other compilation steps. Its a pretty widely known issue and one of the few major problems people have with Rust based wasm. Typically it takes 1-2 minutes, and I have no idea about speeding it up or anything.